code/game/objects/items/weapons/weapon.dm

```diff
@@ -10,3 +9,178 @@
 	var/max_shots = 0 //codex, energy weapons
 	var/scope_zoom = FALSE//codex
 	var/self_recharge = FALSE //codex
+
+	///Bayonet Charge base obj
+	var/datum/action/bayonet_charge/charge
+//its here because if I dont place it in the same script as the base weapon define, it breaks.
+/datum/action/bayonet_charge
+	name = "Toggle Bayonet Charge"
+	action_icon_state = "charge_human"
+
+	//these variables modify the stats of a bayonet charge
+	///the amount of tiles you need to walk to start charging
+	var/minactivationrundist = 5
+	///thhe amount of tiles you need to walk to charge fully
+	var/fullspeedrundist = 9
+	///speed increase per step once charging
+	var/speedperstep = 0.2
+	///MAXIMUMSPEED INCREASE
+	var/maxspeed = 2.4
+	///the amount by which the stam loss is multiplied by per step taken when charging
+	var/stamlossmult = 1
+	///the multiplier for damage inflicted when you bayonet charge someone
+	var/damagemult = 1.5
+	///This is a speed mult for when you finish charging
+	var/postspeedmult = 1.5
+	///the amount of time the post charge speed boost lasts (in deciseconds)
+	var/postspeedtime = 50
+	///the amount of time the stun lasts in 1/10ths of a second
+	var/slownesstime = 5
+	///the amount of time cooldown lasts in deciseconds
+	var/cooldowntime = 150
```

Comment:
for each of those vars, you need to ask yourself, will i ever make a charge using different values.

If yes, you an keep the var. If no, consider changing it for a define